internal/utils/ctrlcreader.go

```diff
@@ -0,0 +1,62 @@
+// Copyright 2020-2021 Authors of Cilium
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package utils
+
+import (
+	"io"
+	"sync"
+)
+
+// CtrlCReader implements a simple Reader/Closer that returns Ctrl-C and EOF
+// on Read() after it has been closed, and nothing before it.
+type CtrlCReader struct {
+	once sync.Once
+	done chan struct{}
+}
+
+// NewCtrlCReader returns a new CtrlCReader instance
+func NewCtrlCReader() *CtrlCReader {
+	return &CtrlCReader{
+		done: make(chan struct{}),
+	}
+}
+
+// Read implements io.Reader.
+// Blocks until we are done.
+func (cc *CtrlCReader) Read(p []byte) (n int, err error) {
+	select {
+	case <-cc.done:
+		if len(p) > 0 {
+			p[0] = byte(3) // Ctrl-C
```

Comment:
Is this ASCII 3? (ETX - end of text) How did you find out which value to send? :)

Reply:
Googling around, found someone sending byte(3) and byte(4); took it from there. Double-checked with an ASCII table and it seemed to make sense + trial and error. `cilium monitor` did not need the byte(4), but recalled that on terminal ctrl-D is a hard exit (I kick myself in the foot every time I do that by accident and lose weeks worth of usable shell history :-).